python/ray/util/worker_group.py

```diff
@@ -0,0 +1,308 @@
+import socket
+from dataclasses import dataclass
+import logging
+from typing import Callable, List, TypeVar, Optional, Dict, Type, Tuple
+
+import ray
+from ray.actor import ActorHandle
+from ray.types import ObjectRef
+
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
+
+
+class BaseWorkerMixin:
+    """A class to execute arbitrary functions. Does not hold any state."""
+
+    def __execute(self, func: Callable[..., T], *args, **kwargs) -> T:
+        """Executes the input function and returns the output.
+
+        Args:
+            func (Callable): The function to execute.
+            args, kwargs: The arguments to pass into func.
+        """
+        return func(*args, **kwargs)
+
+
+@dataclass
+class WorkerMetadata:
```

Comment:
The `WorkerMetadata` and `Worker` data class are also pretty specific to SGD. I can remove those from here

Reply:
yep!

Reply:
Actually can you validate this with other potential library owners before taking this out completely? 

I do agree that the `gpu_ids` field is probably too specific, but collecting general node/Ray worker information might have value for coordination purposes and I think this pattern is helpful to prevent having to make additional blocking calls upstream to fetch more data from the workers throughout their lifecycle.

One idea is to:
1. Generic data class with the most common (or 0) attributes.
2. Allow for consumers to extend the data class with new attributes + data fetching logic.

Reply:
Actually, no need for unstructured dict, the underlying class could return structured data if they wanted to. Let's do something low-friction for now but I'd prefer to keep things generic if we can!